### PR TITLE
feat(web): Discover → register-as-datasource CTA (closes #207)

### LINKED ISSUE #207: DiscoverResultBanner: add CTA to register inferred.prometheusUrl as a datasource
## Context

After PR #199, \`inferred.prometheusUrl\` is kept in the contract intentionally so Discover can still surface the metric endpoint — but there's no UX path from "discovered" to "registered as a Prometheus datasource". User has to manually copy the URL → navigate to /settings/prometheus-datasources → click "+ 新增数据源" → paste.

Surveyed in PR #205 (dropped from that quick-win batch) — turns out to be an **architectural mismatch**:

- \`DiscoverResultBanner\` only mounts when \`countFilledFields === 0\` (zero-results diagnostic path). When \`prometheusUrl.value\` is non-null, Discover hits the auto-apply path — banner doesn't render.
- So a CTA can't live in the banner. Needs to live on the auto-apply success path.

## Proposed design directions

1. **Toast action button** — success toast carries "+ 注册为数据源" action that opens DatasourceSheet pre-populated with the URL.
2. **Inline hint below the form** — dismissible pill below ConnectionSheet's metric section when there's an unregistered inferred Prometheus URL.
3. **Cross-page deep link** — navigate to \`/settings/prometheus-datasources?baseUrl=<inferred>\` and have the page open the create sheet pre-filled.

Recommend direction 1 — minimal new UI, action at the natural moment.

## Out of scope

- Auto-registration without user confirm (security).
- Bind to current connection on register — separate concern (\`set_connection_prometheus_source\` already covers).

## Acceptance

- After Discover success with \`inferred.prometheusUrl.value\` present, user sees an actionable affordance.
- Click → DatasourceSheet opens pre-populated (URL + name suggestion from baseUrl host).
- Test on the auto-apply path asserts the CTA appears.

### PR DESCRIPTION
**Pull request by weetime**
## Summary

After Connection Discover infers a Prometheus URL that's not already a registered datasource, the user (if admin) now sees an inline pill below the `prometheusDatasourceId` Select. Click → `DatasourceSheet` opens pre-populated (URL + `name = new URL(url).host`) → on save, the new datasource id is auto-bound to the Connection form with `shouldDirty: true`.

## Direction

Spec picked **Direction 2 (inline pill)** over the issue's recommended Direction 1 (toast action button): the auto-apply toast TTL is ~4-5s, but users typically scan the form for much longer right after Discover fires, so a persistent contextual pill catches them where their eyes already are. Full reasoning in `docs/superpowers/specs/2026-05-20-discover-register-datasource-cta-design.md`.

## Visibility rule

Pill renders only when **all four** conditions hold:
1. Discover returned a `inferred.prometheusUrl.value`
2. No existing datasource matches that `baseUrl`
3. Connection form's `prometheusDatasourceId` is unset
4. Current user has `admin` role (backend rejects non-admin with 403)

## What's in the PR

- **Spec + plan**: `docs/superpowers/specs/2026-05-20-discover-register-datasource-cta-design.md`, `docs/superpowers/plans/2026-05-20-discover-register-datasource-cta.md`.
- **New utility**: `deriveDatasourceNameFromUrl(url)` — pure URL→host, empty on parse fail. 6 unit tests.
- **DatasourceSheet widening**: `create` mode now accepts an optional `initial?: Partial<DatasourceInput>`. `DatasourceInput` is now exported. 2 new tests (pre-fill + no-initial regression).
- **ConnectionSheet changes**: new `inferredPrometheusUrl` state populated on every Discover (both auto-apply and zero-result paths), new `registerSheetOpen` state, `showRegisterCta` visibility computation, pill JSX inside the existing `prometheusDatasourceId` FormItem, sibling `<DatasourceSheet>` with `onSaved` that auto-selects + clears local state. Return wrapped in Fragment to host the sibling Sheet. 6 new tests covering pill visibility (4 edge cases) + sheet wiring (open with initial / onSaved binds).
- **i18n**: `dialog.discover.registerCta.{headline,body,action}` in zh-CN + en-US.
- **Test-only side fix**: DiagnosticsPage's prometheus-datasources mock had to gain the three datasource mutation hooks (DatasourceSheet is now always mounted whenever ConnectionSheet renders); applied the same defensive stub to ConnectionsPage as a follow-up consistency patch.
- **biome auto-fix**: Fragment wrap shifted indentation in ConnectionSheet.tsx — committed as a separate \`chore(web): biome auto-fix\` so reviewers can skip it visually. Whitespace-ignoring diff for that commit is ~37 lines.

## Test plan

- [x] \`pnpm -F @modeldoctor/web exec vitest run src/features/prometheus-datasources/derive-name.test.ts\` — 6/6 pass
- [x] \`pnpm -F @modeldoctor/web exec vitest run src/features/prometheus-datasources/DatasourceSheet.test.tsx\` — existing + 2 new pass
- [x] \`pnpm -F @modeldoctor/web exec vitest run src/features/connections/ConnectionSheet.test.tsx\` — existing + 6 new (1 happy-path + 4 edge-cases + 1 onSaved) pass
- [x] \`pnpm -F @modeldoctor/web test\` → 852/852 pass
- [x] \`pnpm -F @modeldoctor/web type-check\` → clean
- [x] \`pnpm -F @modeldoctor/web lint\` → clean (biome + i18n parity + no-native-select + no-confirm + no-handcrafted-popover-list)
- [ ] Manual: admin user, ConnectionSheet → paste curl with a Prometheus URL not in datasources → pill appears under "Prometheus 数据源" → click → DatasourceSheet opens pre-filled → save → Connection's Prometheus Select shows the new datasource selected

Closes #207.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/web/src/features/connections/ConnectionSheet.tsx [hunk 1]
```diff
@@ -447,616 +455,676 @@ export function ConnectionSheet({
...
 
+  // Discover → register CTA — see issue #207. Show the pill only when all
+  // four conditions hold: an inferred URL exists, it's not already a
+  // registered datasource, the user hasn't picked any datasource yet, and
+  // the current user is an admin (backend requires admin for create).
+  const user = useAuthStore((s) => s.user);
+  const isAdmin = (user?.roles ?? []).includes("admin");
+  const watchedDsId = form.watch("prometheusDatasourceId");
+  const inferredAlreadyRegistered = inferredPrometheusUrl
+    ? (datasources ?? []).some((d) => d.baseUrl === inferredPrometheusUrl)
```
[the remark this answers is not in the document]
Comment (weetime) [reply apps/web/src/features/connections/ConnectionSheet.tsx]: Thanks — addressed in c749442.

Went slightly further than the inline regex+lowercase suggestion: extracted a `normalizeBaseUrl(url)` utility next to `derive-name.ts` that lowercases only the protocol+host (HTTP path is case-sensitive per RFC 3986), strips trailing slashes, and preserves query/hash. 10 unit tests for the util + 1 behavioral test in ConnectionSheet asserting the pill hides when the inferred URL differs from a registered one only by trailing slash + scheme case.
